_articles/fr/2023-10-25-atomic-design.md

```diff
@@ -0,0 +1,210 @@
+
+## Hierarchie des éléments
+#### Atomes
+Ici, nous allons classer les éléments les plus petits et indivisibles. Il s'agit généralement d'éléments HTML tels que des liens, des boutons, des champs de texte, des images, etc. On créera un composant distinct pour chacun de ces éléments atomiques. Les atomes sont par essence des composants abstraits et ne possèdent pas intrinsèquement une grande utilité.
+
+Dans une approche purement axée sur l'interface utilisateur, on crée un atome distinct pour chaque spécificité. Ainsi, un bouton bleu est considéré comme un atome distinct de celui d'un bouton rouge. Cependant, dans une perspective de programmation, il est plus courant de créer un atome unique qui prend en compte des arguments pour faire évoluer ses attributs en fonction des besoins.
+
+```jsx
+// Atom - Button.tsx
+const Button = ({ label, color, handleClick }) => (
+  <button style={{ backgroundColor: color }} onClick={handleClick} >{label}</button>
+);
+
+export default Button;
+```
+
+#### Molécules
+Comme en chimie, une molécule est composée d'un ensemble d'atomes qui forment les éléments de base de l'interface utilisateur. Par exemple, un composant SearchBar est composé de l'atome `Input` et de l'atome `Button`.
+
+Si vous vous trouvez dans une situation où vous devez inclure un élément HTML dans une molécule, cela indique qu'un atome distinct doit être créé pour cet élément.
+
+```jsx
+// Molecule - SearchBar.tsx
+import Button from './Atoms/Button';
+import Input from './Atoms/Input';
+import Label from './Atoms/Label';
+
+const SearchBar = ({ searchBarLabel, searchBarInputPlaceholder, searchBarButtonProps }) => {
+
+  // ❌ NOT TO DO
+  /*return (
+    <div>
+      <Label content="Search in site" />
+      <Input type="text" placeholder="Type your search" />
+      <Button label="Search"  />
+    </div>
+  );*/
+
+  // ✅ TO DO
+  return (
+    <div>
+      <Label content={searchBarLabel} />
+      <Input type="text" placeholder={searchBarInputPlaceholder} />
+      <Button { ...searchBarButtonProps} />
+    </div>
+  );
+}
+
+export default SearchBar;
+```
+
+Dans cet exemple, le composant `SearchBar` prend en argument les propriétés `searchBarLabel`, `searchBarInputPlaceholder`, `searchBarButtonProps`, qu'il transmet respectivement aux atomes `Label`, `Input` et `Button`. L'erreur serait de fournir ces informations depuis ce niveau, car notre objectif est de maintenir nos composants aussi génériques que possible, sans inclure des spécificités à ce stade.
```

Comment:
Une idée pour ton contenu:

```mdx
<div class="admonition failure" markdown="1"><p  class="admonition-title">NOT TO DO</p>

Description du pourquoi
</div>

```tsx
const SearchBar: React.FC = () => (
    <div>
      <Label content="Search in site" />
      <Input type="text" placeholder="Type your search" />
      <Button label="Search"  />
    </div>
  );
```
```